### DISCUSSION TITLE
TV OUT

### DISCUSSION THREAD

**Discussion by josephresearcher** [fdivitto/FabGL#212] [Ideas]
Hello
Dear Fabrizio
Please Add Tv Out(composite)
Thanks

**Comment by fdivitto** [comment]
there is a sponsor goal for this:

[https://github.com/sponsors/fdivitto](https://github.com/sponsors/fdivitto)

I'm working on, anyway to have a good composite output I need very expensive equipment, so I need your help.

**Comment by ghost** [comment]
This repo implements PAL and NTSC https://github.com/rossumur/esp_8_bit

Maybe you could ask if the owner would mind if you used his solutions for FabGL.

**Reply by bratoff** [reply]
Here's a derivative of that one with just the video generation extracted out and wrapped in an Arduino class:
[(https://github.com/Roger-random/ESP_8_BIT_composite)]

**Comment by fdivitto** [comment]
Thanks, but I'm not in a hurry, so I prefer to do things myself.

**Reply by ghost** [reply]
Hey Fabrizio - even if you write the code yourself it's worth reading the readme.md in https://github.com/rossumur/esp_8_bit there's some pretty major insights there on how to generate stable NTSC/PAL signals on esp32.

**Reply by fdivitto** [reply]
Actually there are a lot of books and tech references I've already read about composite video. I'm working on it since last year! 
I had to buy two waveform/vectorscope instruments (a Tektronix VM400A and a Rohde & Schwarz VSA), because I wanted a video generator almost perfectly compliant with PAL and NTSC specifications (very different by other implementations). I don't know if I will succeed in my intent.

if you want to know the progress follow this:

[https://github.com/fdivitto/FabGL/discussions/192](https://github.com/fdivitto/FabGL/discussions/192)

**Comment by josephresearcher** [comment]
Temporary solution with AD724 

https://electronics.stackexchange.com/questions/403637/vga-to-composite-circuit-does-it-work

**Comment by fdivitto** [comment]
please follow this:

[https://github.com/fdivitto/FabGL/discussions/192](https://github.com/fdivitto/FabGL/discussions/192)
